frank/fit.py

```diff
@@ -364,145 +359,164 @@ def output_results(u, v, vis, weights, sol, iteration_diag, iter_plot_range,
-                                       iteration_diag['num_iterations']]
-
-        diag_fig, diag_axes = make_figs.make_diag_fig(sol.r, sol.q,
-                                                      iteration_diag,
-                                                      iter_plot_range,
-                                                      force_style, save_prefix
-                                                      )
+    if model['plotting']['diag_plot']:
+        diag_fig, diag_axes, iter_plot_range = make_figs.make_diag_fig(sol.r, sol.q,
+                                                                       iteration_diagnostics,
+                                                                       model['plotting']['iter_plot_range'],
+                                                                       model['plotting']['force_style'],
+                                                                       model['input_output']['save_prefix']
+                                                                       )
+
+        model['plotting']['iter_plot_range'] = iter_plot_range
 
         figs.append(diag_fig)
         axes.append(diag_axes)
 
-    logging.info('  Saving results')
-
-    io.save_fit(u, v, vis, weights, sol, save_prefix,
-                save_profile_fit, save_vis_fit, save_uvtables,
-                save_iteration_diag, iteration_diag,
-                format=output_format
+    io.save_fit(u, v, vis, weights, sol,
+                model['input_output']['save_prefix'],
+                model['input_output']['save_solution'],
+                model['input_output']['save_profile_fit'],
+                model['input_output']['save_vis_fit'],
+                model['input_output']['save_uvtables'],
+                model['input_output']['iteration_diag'],
+                iteration_diagnostics,
+                model['input_output']['format']
                 )
 
-    return figs, axes
+    return figs, axes, model
+
+
+def perform_bootstrap(u, v, vis, weights, geom, model):
+    r"""
+    Perform a bootstrap analysis for the Franktenstein fit to a dataset
+
+    Parameters
+    ----------
+    u, v : array, unit = :math:`\lambda`
+        u and v coordinates of observations
+    vis : array, unit = Jy
+        Observed visibilities (complex: real + imag * 1j)
+    weights : array, unit = Jy^-2
+        Weights assigned to observed visibilities, of the form
+        :math:`1 / \sigma^2`
+    geom : SourceGeometry object
+        Fitted geometry (see frank.geometry.SourceGeometry)
+    model : dict
+        Dictionary containing model parameters the fit uses
+
+    Returns
+    -------
+    boot_fig : Matplotlib `.Figure` instance
+        The produced figure, including the GridSpec
+    boot_axes : Matplotlib `~.axes.Axes` class
+        The axes of the produced figure
+    """
+    profiles_bootstrap = []
+
+    for ii in range(model['analysis']['n_trials']):
```

Comment:
Rename ii to something more informative, e.g. trial

```suggestion
    for trial in range(model['analysis']['n_trials']):
```